python/tk_multi_loader/dialog.py

```diff
@@ -1099,11 +1101,41 @@ def _load_entity_presets(self):
 
         for e in entities:
 
-            # validate that the settings dict contains all items needed.
-            for k in ["caption", "entity_type", "hierarchy", "filters"]:
+            # Validate that the settings dictionary contains all items needed.
+
+            k_error_msg = "Configuration error: One or more items in %s are missing a '%s' key!"
+            v_error_msg = "Configuration error: One or more items in %s have an invalid '%s' value for '%s' key!"
+
+            k = "caption"
+            if k not in e:
+                raise TankError(k_error_msg % (entities, k))
+
+            preset_name = e["caption"]
```

Comment:
It would just mean duplicating the `'caption'` validation and would not really create two fully separate branches since the `'type'` has to be tested on its own before branching. Using better variable names already made the code easier to read.